### PR TITLE
feat(authz/ldap): Add LDAP role provider support for authorization.

### PR DESCRIPTION
**Pull request by stewchen**

### CODE REVIEW

File: halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/security/authz/ldap/EditLdapRoleProviderCommand.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  @Parameter(
+      names = "--url",
+      description = "ldap:// or ldaps:// url of the LDAP server",
+      converter = URIConverter.class
+  )
+  private URI url;
+
+  @Parameter(
+      names = "--manager-dn",
+      description = "The manager's user distinguished name (principal) to use for querying ldap groups."
```
Comment (ttomsu) [inline halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/security/authz/ldap/EditLdapRoleProviderCommand.java]: I think it should be "manager user's"

File: halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/security/authz/ldap/EditLdapRoleProviderCommand.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+  )
+  private String groupRoleAttributes;
+
+  @Parameter(
+      names = "--user-dn-pattern",
+      description = "The pattern for finding a user's DN using simple pattern matching. For "
+          + "example, if your LDAP server has the URL ldap://mysite.com/dc=spinnaker,dc=org, "
+          + "and you have the pattern 'uid={0},ou=members', 'me' will map to a DN "
+          + "uid=me,ou=members,dc=spinnaker,dc=org. If no match is found, will try to find the user "
+          + "using user-search-filter, if set."
```
Comment (ttomsu) [inline halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/security/authz/ldap/EditLdapRoleProviderCommand.java]: Add two dashes before `user-search-filter`

File: halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/security/authz/ldap/EditLdapRoleProviderCommand.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+  )
+  private String userDnPattern;
+
+  @Parameter(
+      names = "--user-search-base",
+      description = "The part of the directory tree under which user searches should be performed. "
+          + "If user-search-base isn't supplied, the search will be performed from the root."
```
Comment (ttomsu) [inline halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/security/authz/ldap/EditLdapRoleProviderCommand.java]: Add two dashes

File: halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/security/authz/ldap/EditLdapRoleProviderCommand.java [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+  )
+  private String userSearchBase;
+
+  @Parameter(
+      names = "--user-search-filter",
+      description = "The filter to use when searching for a user's DN. Will search either from "
+          + "user-search-base (if specified) or root for entires matching the filter, then attempt "
+          + "to bind as that user with the login password. For example, the filter 'uid={0}' would "
```
Comment (ttomsu) [inline halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/security/authz/ldap/EditLdapRoleProviderCommand.java]: > then attempt to bind as that user with the login password.

I don't think this is correct. The manager username/password is used to establish the connection (bind) to the LDAP server, but AFAIK the user's password is not used at all when loading groups/roles from an LDAP server.

File: halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/security/authz/ldap/EditLdapRoleProviderCommand.java (hunk 1 above)
Comment (stewchen) [reply halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/security/authz/ldap/EditLdapRoleProviderCommand.java]: Done.

File: halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/security/authz/ldap/EditLdapRoleProviderCommand.java (hunk 3 above)
Comment (stewchen) [reply halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/security/authz/ldap/EditLdapRoleProviderCommand.java]: Done

File: halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/security/authz/ldap/EditLdapRoleProviderCommand.java (hunk 4 above)
Comment (stewchen) [reply halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/security/authz/ldap/EditLdapRoleProviderCommand.java]: Removed

**Pull request merged**